sdk/state/update_test.go

```diff
@@ -0,0 +1,348 @@
+			ResponderSigner: responder.KP.FromAddress(),
+			InitiatorEscrow: initiator.Escrow.FromAddress(),
+			ResponderEscrow: responder.Escrow.FromAddress(),
+			StartSequence:   s,
+		})
+		require.NoError(t, err)
+		f, err = f.Sign(networkPassphrase, initiator.KP, responder.KP)
+		require.NoError(t, err)
+		fbtx, err := txnbuild.NewFeeBumpTransaction(txnbuild.FeeBumpTransactionParams{
+			Inner:      f,
+			FeeAccount: initiator.KP.Address(),
+			BaseFee:    txnbuild.MinBaseFee,
+		})
+		require.NoError(t, err)
+		fbtx, err = fbtx.Sign(networkPassphrase, initiator.KP)
+		require.NoError(t, err)
+		_, err = client.SubmitFeeBumpTransaction(fbtx)
+		require.NoError(t, err)
+	}
+
+	// TODO - create NewChannel method - always sets Balance to 0 at first
+	initiatorChannel := Channel{
+		Status:                 "initialized",
+		ProposalStatus:         "none",
+		Initiator:              true,
+		InitiatorEscrowAccount: initiator.Escrow.FromAddress(),
+		ResponderEscrowAccount: responder.Escrow.FromAddress(),
+		Balance:                0,
+		Asset:                  txnbuild.NativeAsset{},
+	}
+
+	responderChannel := Channel{
+		Status:                 "initialized",
+		ProposalStatus:         "none",
+		Initiator:              false,
+		InitiatorEscrowAccount: initiator.Escrow.FromAddress(),
+		ResponderEscrowAccount: responder.Escrow.FromAddress(),
+		Balance:                0,
+		Asset:                  txnbuild.NativeAsset{},
+	}
+
+	//// SETUP DONE
+
+	//// NEW PROPOSALS
+	paymentProposal := PaymentProposal{}
+	for i < 7 {
+		i++
+
+		amount := randomPositiveInt64(t, 100_0000000)
+		amountToResponder := int64(0)
+		amountToInitiator := int64(0)
+		paymentLog := ""
+		if randomBool(t) {
+			amountToResponder = amount
+			paymentLog = "I payment to R of: "
+		} else {
+			amountToInitiator = amount
+			paymentLog = "R payment to I of: "
+		}
+		t.Log("Proposal: ", i, paymentLog, amount)
+
+		//// INITIATOR: creates new Payment, sends to R
+		paymentProposal, err := initiatorChannel.NewPaymentProposal(initiator, responder, amountToInitiator, amountToResponder, s, i, e, 0, 0, networkPassphrase)
+		require.NoError(t, err)
+		j, err := json.Marshal(paymentProposal)
+		require.NoError(t, err)
+
+		//// RESPONDER: receives new payment proposal, validates, then confirms by signing both
+		paymentProposal = &PaymentProposal{}
+		err = json.Unmarshal(j, paymentProposal)
+		require.NoError(t, err)
+		paymentProposal, err = responderChannel.ConfirmPayment(paymentProposal, initiator, responder, networkPassphrase)
+		require.NoError(t, err)
+		j, err := json.Marshal(paymentProposal)
+		require.NoError(t, err)
+
+		//// INITIATOR: re-confirms P_i by signing D_i and sending back
+		paymentProposal = PaymentProposal{}
+		err = json.Unmarshal(j, &paymentProposal)
+		paymentProposal, err = initiatorChannel.ConfirmPayment(paymentProposal, networkPassphrase)
+		require.NoError(t, err)
+	}
+
+	// TODO - how does Initiator submit "latest", is it store in the Channel?
```

Comment:
Good question, I think we'll store in channel, or rather the tx can regenerate the tx and add the signatures.